File: src/main/java/com/mttnow/system/team/model/Person.java
package com.mttnow.system.team.model;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.constraints.NotEmpty;

@Document
@Data
@EqualsAndHashCode
public class Person {

  @Id
  String id;

  @NotEmpty
  String firstName;
  String lastName;
  int age;

}

File: src/main/kotlin/com/mttnow/system/team/model/Person.kt
package com.mttnow.system.team.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

import javax.validation.constraints.NotEmpty

@Document
data class Person(
    @field:Id var id: String? = "",       //If not 'field:' on annotation, applied to constructor only, not at a field level!
    @field:NotEmpty var firstName: String? = "",
    var lastName: String? = "",
    var age: Int = 0)
